module2-database-technologies/README.md

```diff
@@ -1 +1,372 @@
-# curriculum-backend-readings
\ No newline at end of file
+# Module 2: Databases
+
+## Introduction to Databases
+In this module, you'll learn about **databases**. Database is a general term
+that refers to the data stored in a structured format. We use databases when we
+want to persist data for future use. For example, in a web application, when a
+user writes something in a text box, this data will be lost when the page is
+refreshed -- unless this data is persisted in a database.
+
+Databases are a ubiquitous and foundational part of backend technology.
+
+### Introduction to relational databases
+
+One of the most common types of databases is the **relational database**. A
+relational database stores data in tables, and these tables may have *relations*
+to one another. For example, you may have a table `Customer`; each row
+represents a customer in your application. You may have another table 
+`Order`, a customer can order many things, and each order will appear in this
+table. We will later look in more detail at how the two tables are linked.
+
+Relational databases are frequently referred to as SQL databases. SQL is a
+[declarative programming
+language](https://365datascience.com/tutorials/sql-tutorials/sql-declarative-language/)
+that is commonly used to query for data. Not all relational databases
+necessarily use SQL, but it is by far the most common, so in practice, SQL
+database and relational database are synomyous terms.
+
+The most frequently used relational databases are PostgreSQL and MySQL. There
+are many offerings for SQL databases, and each may have slightly different
+syntax for SQL. However, if you understand the general concept of relational
+databases, you will be able to easily use different databases. 
+
+### Understanding relational database structure
+As mentioned previously, data in relational databases are usually organized in a
+tabular format, similar to spreadsheets.
+
+* A database has many tables
+* A table has many rows (also known as records)
+* A row has many columns (also known as fields).
+
+For example, the previously mentioned `Customer` table may look something like
+the following. Note that this is not any particular syntax but simply a
+visualization of the data.
+
+**id**|**first\_name**|**last\_name**|**registered\_at**
+:-----:|:-----:|:-----:|:-----:
+1|Ammar|Sammour|2012-01-02
+2|Halit|Batur|2012-01-03
+3|Shrreya|Bhatachaarya|2012-01-05
```

Comment:
Maybe it would be better to add generic names here, since the curriculum will be used eventually across multiple cohorts.